doctr/models/recognition/vitstr/pytorch.py

```diff
@@ -0,0 +1,237 @@
+            return out
+
+        if return_model_output:
+            out["out_map"] = decoded_features
+
+        if target is None or return_preds:
+            # Post-process boxes
+            out["preds"] = self.postprocessor(decoded_features)
+
+        if target is not None:
+            out["loss"] = self.compute_loss(decoded_features, gt, seq_len)
+
+        return out
+
+    @staticmethod
+    def compute_loss(
+        model_output: torch.Tensor,
+        gt: torch.Tensor,
+        seq_len: torch.Tensor,
+    ) -> torch.Tensor:
+        """Compute categorical cross-entropy loss for the model.
+        Sequences are masked after the EOS character.
+        Args:
+            model_output: predicted logits of the model
+            gt: the encoded tensor with gt labels
+            seq_len: lengths of each gt word inside the batch
+        Returns:
+            The loss of the model on the batch
+        """
+        # Input length : number of timesteps
+        input_len = model_output.shape[1]
+        # Add one for additional <eos> token
+        seq_len = seq_len + 1
+        # Compute loss
+        # (N, L, vocab_size + 1)
+        cce = F.cross_entropy(model_output.permute(0, 2, 1), gt, reduction="none")
+        mask_2d = torch.arange(input_len, device=model_output.device)[None, :] >= seq_len[:, None]
+        cce[mask_2d] = 0
+
+        ce_loss = cce.sum(1) / seq_len.to(dtype=model_output.dtype)
+        return ce_loss.mean()
+
+
+class ViTSTRPostProcessor(RecognitionPostProcessor):
+    """Post processor for ViTSTR architectures
+
+    Args:
+        vocab: string containing the ordered sequence of supported characters
+    """
+
+    def __call__(
+        self,
+        logits: torch.Tensor,
+    ) -> List[Tuple[str, float]]:
+        # compute pred with argmax for attention models
+        out_idxs = logits.argmax(-1)
+        # N x L
+        probs = torch.gather(torch.softmax(logits, -1), -1, out_idxs.unsqueeze(-1)).squeeze(-1)
+        # Take the minimum confidence of the sequence
+        probs = probs.min(dim=1).values.detach().cpu()
+
+        # Manual decoding
+        word_values = [
+            "".join(self._embedding[idx] for idx in encoded_seq).split("<eos>")[0]
+            for encoded_seq in out_idxs.detach().cpu().numpy()
+        ]
+
+        return list(zip(word_values, probs.numpy().tolist()))
+
+
+def _vitstr(
+    arch: str,
+    pretrained: bool,
+    pretrained_backbone: bool = True,
+    ignore_keys: Optional[List[str]] = None,
+    **kwargs: Any,
+) -> ViTSTR:
+
+    pretrained_backbone = pretrained_backbone and not pretrained
+
+    # Patch the config
+    _cfg = deepcopy(default_cfgs[arch])
+    _cfg["vocab"] = kwargs.get("vocab", _cfg["vocab"])
+    _cfg["input_shape"] = kwargs.get("input_shape", _cfg["input_shape"])
+
+    kwargs["vocab"] = _cfg["vocab"]
+    kwargs["input_shape"] = _cfg["input_shape"]
+
+    # Feature extractor
+    feat_extractor = vit(
```

Comment:
@frgfm that's what i mean we want use the pretrained model but need to init it with different args.. i have not really an idea how we can reach this with IntermediateLayerGetter

Reply:
well we can use `IntermediateLayerGetter` and take the last layer :)

Reply:
we have to init ViT with completly other args ... i dont know how this can work :sweat_smile:

Reply:
ViT-B:

```
input_shape: Tuple[int, int, int] = (32, 32, 3),
patch_size: Tuple[int, int] = (4, 4),
d_model: int = 768,
num_layers: int = 12,
num_heads: int = 12,
ffd_ratio: int = 4,
dropout: float = 0.0,
num_classes: int = 1000,
include_top: bool = True,
```

We need:
```
input_shape: Tuple[int, int, int] = (32, 128, 3),
patch_size: Tuple[int, int] = (4, 8),
d_model: int = 384,
num_layers: int = 12,
num_heads: int = 6,
ffd_ratio: int = 4,
dropout: float = 0.0,
include_top: bool = False,
```